shared_model/interfaces/iroha_internal/block_variant.hpp

```diff
@@ -12,10 +12,66 @@
 namespace shared_model {
   namespace interface {
 
-    /// Block variant with either block or empty block
-    using BlockVariantType =
-        boost::variant<std::shared_ptr<shared_model::interface::Block>,
-                       std::shared_ptr<shared_model::interface::EmptyBlock>>;
+    class BlockVariant
+        : public boost::variant<
+              std::shared_ptr<shared_model::interface::Block>,
+              std::shared_ptr<shared_model::interface::EmptyBlock>>,
+          protected AbstractBlock {
+     private:
+      using VariantType =
+          boost::variant<std::shared_ptr<shared_model::interface::Block>,
+                         std::shared_ptr<shared_model::interface::EmptyBlock>>;
+
+     public:
+      using VariantType::VariantType;
+
+      interface::types::HeightType height() const override {
+        return iroha::visit_in_place(
+            *this, [](const auto &any_block) { return any_block->height(); });
+      }
+
+      const interface::types::HashType &prevHash() const override {
+        return std::move(iroha::visit_in_place(
+            *this,
+            [](const auto &any_block) { return any_block->prevHash(); }));
+      }
+
+      const interface::types::BlobType &blob() const override {
+        return std::move(iroha::visit_in_place(
```

Comment:
Please make some private template method which will accept method name and arguments that will pass it to visit in place, so that there is no need to write visit and lambda in every method.